app/models/project.rb

```diff
@@ -18,37 +19,29 @@ class Project < ApplicationRecord
   belongs_to :regional_delivery_officer, class_name: "User", optional: true
 
   def establishment
-    @establishment || retrieve_establishment
```

Comment:
These memoizations will result in making the API call even if the instance variable has a value

I think something like this might work (not tested)

```ruby
def establishment
  @establishment ||= begin
    result = AcademiesApi::Client.new.get_establishment(urn)
    raise result.error if result.error.present?
    establishment = result.object
  end
end

Reply:
Yep, I think you are right! Thank you! 👍